File: 2018/julia/day01.jl
using DelimitedFiles

nums = readdlm("data/input-01.txt", Int)[:, 1]

res = sum(nums)
println("Part 1: $(res)")
@assert(res == 587)

seen, found, cumsum = Dict(), false, 0
while !found
    for i in nums
        global cumsum += i
        if cumsum in keys(seen)
            global found = true
            break
        else
            seen[cumsum] = true
        end
    end
end
println("Part 2: $(cumsum)")
@assert(cumsum == 83130)

File: 2018/nim/day01.nim
import math, sets, strutils, sugar

let nums = collect(newSeq):
    for line in lines("data/input-01.txt"):
        parseInt(line)

var res = sum(nums)
echo "Part 1: ", res
assert(res == 587)

var seen: HashSet[int]
var (found, cumsum) = (false, 0)
while not found:
    for i in nums:
        cumsum += i
        if seen.containsOrIncl(cumsum):
            found = true
            break
echo "Part 2: ", cumsum
assert(cumsum == 83130)
